File: app/src/main/java/libs/com/rogerio/utils/SpriteSimpleInfoData.java
package libs.com.rogerio.utils;

import android.graphics.Bitmap;
import android.graphics.Rect;

public class SpriteSimpleInfoData {
	public int w;
	public int h;
	
	public int x;
	public int y;
	public double speed;
	public boolean kill;
	public float rotate;
	public double deltaX;
	public double deltaY;
	//Trata image
	public int frame;
	public long time=0;
	private Bitmap bmp=null;
	
	private Rect mRect;
	public Bitmap getBmp() {
		return bmp;
	}

	public void setBmp(Bitmap bmp) {
		this.bmp = bmp;
		w = getW();
		h=getH();
	}

	public int getH() {
		if(bmp!=null)
			return bmp.getHeight();
		return h;
	}
	
	public int getW() {
		if(bmp!=null)
			return bmp.getWidth();
		return w;
	}
	
	public Rect getRect() {
		return new Rect(x, y, x+getW(), y+getH());
	}
	
}

File: app/src/main/java/libs/com/rogerio/utils/SpriteSimpleInfoData.kt
package libs.com.rogerio.utils

import android.graphics.Bitmap
import android.graphics.Rect

class SpriteSimpleInfoData {
    var w: Int = 0
    var h: Int = 0

    var x: Int = 0
    var y: Int = 0
    var speed: Double = 0.toDouble()
    var kill: Boolean = false
    var rotate: Float = 0.toFloat()
    var deltaX: Double = 0.toDouble()
    var deltaY: Double = 0.toDouble()
    //Trata image
    var frame: Int = 0
    var time: Long = 0
    var bmp: Bitmap? = null
        set(bmp) {
            field = bmp
            w = getWidth()
            h = getHeight()
        }

    private val mRect: Rect? = null

    val rect: Rect
        get() = Rect(x, y, x + getWidth(), y + getHeight())

    fun getHeight(): Int {
        return if (this.bmp != null) this.bmp!!.height else h
    }

    fun getWidth(): Int {
        return if (this.bmp != null) this.bmp!!.width else w
    }

}